nuxeo-distribution/nuxeo-server-hotreload-tests/src/test/java/org/nuxeo/ftest/server/hotreload/ITDevHotReloadTest.java

```diff
@@ -249,4 +263,60 @@ public void testHotReloadFileImporters() throws IOException {
         assertEquals("Foo", docType);
     }
 
+    @Test
+    /** @since 11.1 **/
+    public void shouldComputeMimeTypeWhenHotReloadDocumentModelSchema() throws IOException {
+        hotReloadRule.deployJarDevBundle(ITDevHotReloadTest.class, "testHotReloadSchema");
+
+        // Creates the document.
+        String docId = RestHelper.createDocument("/", "CustomDocType", "myDocument");
+        Document document = RestHelper.repository().fetchDocumentById(docId);
+
+        // Creates the contents.
+        BatchUploadManager batchUploadManager = RestHelper.batchUploadManager();
+
+        FileBlob image = new FileBlob(Files.createTempFile("", "").toFile(), "anyImage.png");
+        BatchUpload imageBatch = batchUploadManager.createBatch();
+        imageBatch = imageBatch.upload("0", image);
+
+        FileBlob audio = new FileBlob(Files.createTempFile("", "").toFile(), "anySong.mp3");
+        BatchUpload audioBatch = batchUploadManager.createBatch();
+        audioBatch = audioBatch.upload("0", audio);
```

Comment:
You can have a look to `testHotReloadFileImporters` to upload blobs without using java client directly.

Reply:
i agree,  i  looked to the  `testHotReloadFileImporters` before adding the java client directly, but  i thought that this operation create blob on `file:content` perhaps i missed something. i looked to the implementation of this operation but i didn't see a way to pass the fields names that i need in my case the new schema define a new type that hold a simple and a complex type as below :

```
 name="image" type="nxs:content" />
  <xs:element name="question" type="nxs:question-type" />
  <xs:complexType name="question-type">
    <xs:sequence>
      <xs:element name="audio" type="nxs:content" />
    </xs:sequence>
  </xs:complexType>
```

from what i see on perhaps it's possible to do it by adding an adapter to set exactly what i need 
thx

Reply:
Indeed because the `testHotReloadFileImporters` tested explicitly this operation, in your case you could certainly use `Blob.AttachOnDocument` operation (see [java class](https://github.com/nuxeo/nuxeo/blob/master/nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/operations/blob/AttachBlob.java)) which allows you to give as parameter the `xpath`, the `document` and `the blob` as input.